tools/test/test_codegen.py

```diff
@@ -270,6 +276,35 @@ def test_native_function_declaration_1_op_1_ns_valid(self) -> None:
         self.assertEqual("\n".join(declaration), target)
 
 
+# Test for native_function_generation
+class TestNativeFunctionGeneratrion(unittest.TestCase):
+    def setUp(self) -> None:
+        self.native_functions: List[NativeFunction] = []
+        self.backend_indices: Dict[
+            DispatchKey, Dict[OperatorName, BackendMetadata]
+        ] = defaultdict(dict)
+        yaml_entry = """
+- func: op(Tensor self) -> Tensor
+  dispatch:
+    CompositeExplicitAutograd: op
+  autogen: op.out
+        """
+        es = yaml.load(yaml_entry, Loader=LineLoader)
+        func, m = NativeFunction.from_yaml(
+            es[0], loc=Location(__file__, 1), valid_tags=set()
+        )
+        self.native_functions.append(func)
+        BackendIndex.grow_index(self.backend_indices, m)
+
+    def test_functional_variant_autogen_out_variant(self) -> None:
+        add_generated_native_functions(self.native_functions, self.backend_indices)
+        self.assertEqual(len(self.native_functions), 2)
```

Comment:
it might be useful for this to test an edge case: e.g. if the schema of `op` were `op(Tensor self) -> (Tensor, Tensor)`, then we'd expect the out= argument of the generated function to be a `Tuple[Tensor, Tensor]`.